dedupe/variables/ilcs.py

```diff
@@ -0,0 +1,22 @@
+import ilcs_parser
+from parseratorvariable import ParseratorType
+
+CITATION = (
+    ('chapter', ('Chapter',)),
+    ('act prefix', ('ActPrefix')),
+    ('section', ('Section')),
+    ('subsection', ('SubSection')),
+    ('attempted', ('Attempted'))
+)
+
+
+class ILCSType(ParseratorType):
+    type = 'ILCS'
+
+    def tagger(self, field):
+        return self.tag(field)
+
+    def __init__(self, definition):
+        self.components = (('Citation', self.compareFields, CITATION),)
```

Comment:
I think I probably want to extend `compareFields` to try to compare each token as a sequential number/letter instead of as a string. Perhaps what I want to do is override the code block linked below so that instead of calling `compareString` I call a custom comparison function?

https://github.com/dedupeio/parseratorvariable/blob/bb8808b05418300bcf42d88bc4e8570a65011ffa/parseratorvariable/__init__.py#L161-L164

Reply:
exactly! this is compareFields for the name comparator https://github.com/dedupeio/dedupe-variable-name/blob/master/dedupe/variables/name.py#L157